Modules/posixmodule.c

```diff
@@ -7196,22 +7196,21 @@ os_sched_getaffinity_impl(PyObject *module, pid_t pid)
 #  define DEV_PTY_FILE "/dev/ptmx"
 #endif
 
-#if defined(HAVE_OPENPTY) || defined(HAVE_FORKPTY) || defined(HAVE_DEV_PTMX)
-#ifdef HAVE_PTY_H
+#if defined(HAVE_OPENPTY) || defined(HAVE_FORKPTY) || defined(HAVE_LOGIN_TTY) || defined(HAVE_DEV_PTMX)
+#if defined(HAVE_PTY_H)
```

Comment:
We prefer the shorter ``ifdef`` when it can be used.

```suggestion
#ifdef HAVE_PTY_H
```

Reply:
@tiran I just realized that I had missed some of the `#if defined()`. Now it is fixed; I have changed all of them to `#ifdef`. However, there is a merge conflict now. I am working on fixing that.